universe/archive.go

```diff
@@ -711,6 +716,21 @@ func (a *Archive) FetchLeaves(ctx context.Context,
 	)
 }
 
+// UpsertIgnoreTuples inserts the passed ignore tuples into the ignore tree
+// archive. If the tuples already exist, then no action is taken.
+//
+// NOTE: Ignore proofs are not included in the multiverse, as they are purely
+// off-chain. Only issuance and transfer proofs are included in the multiverse.
+func (a *Archive) UpsertIgnoreTuples(ctx context.Context,
```

Comment:
In the context of the greater system, we wouldn't want to insert ignore tuples directly into the tree. We want to make sure that we're always serving information that's consistent with the latest commitment on chain. Hence the way we stage updates as state transitions in #1508.